### PR TITLE
Feat/cqc approval

### PR DESCRIPTION
**Pull request by g-m-taylor**
everything needed for the CQC - Admin can approve or reject a change of main service to a CQC-regulated service

#### Tests
Does this PR include tests for the changes introduced?
- [x ] Yes
- [ ] No, I found it difficult to test
- [ ] No, they are not required for this change.

### CODE REVIEW

File: src/app/features/search/cqc-status-changes/cqc-status-changes.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
+// import { SharedModule } from '@shared/shared.module';
```
Comment (g-m-taylor) [inline src/app/features/search/cqc-status-changes/cqc-status-changes.component.spec.ts:1]: Tests commented out as they will used in the next PR. ( sorry shouldnt have uploaded them in the first palce)

File: src/app/features/search/cqc-status-change/cqc-status-change.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,300 @@
+// import { SharedModule } from '@shared/shared.module';
```
Comment (g-m-taylor) [inline src/app/features/search/cqc-status-change/cqc-status-change.component.spec.ts:1]: Tests commented out as they will used in the next PR. ( sorry shouldnt have uploaded them in the first palce)

File: server/routes/admin/cqc-status-change/index.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+  } catch (error) {
+    console.error(error);
+    return res.status(400).send();
+  }
+};
+
+const _mapResults = async (approvalResults) => {
+  const promises = approvalResults.map(async approval => {
+    data = approval.Data;
+    console.log(data);
```
Comment (chadtomkiss-madetech) [inline server/routes/admin/cqc-status-change/index.js:39]: console.log ;(

File: server/routes/admin/cqc-status-change/index.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        other: data.requestedService.other || null
+      }
+    };
+  });
+  return await Promise.all(promises);
+};
+
+const _approveChange = async (req, res) => {
+  console.log('_approveChange');
+  console.log(req.body.approvalId);
```
Comment (chadtomkiss-madetech) [inline server/routes/admin/cqc-status-change/index.js:70]: console.log ;(

File: server/routes/admin/cqc-status-change/index.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+  await _updateApprovalStatus(req.body.approvalId, 'Approved');
+  const results = await _updateMainService(req.body.approvalId, req.username);
+  console.log('results:');
+  console.log(results);
```
Comment (chadtomkiss-madetech) [inline server/routes/admin/cqc-status-change/index.js:74]: console.log ;(

File: server/routes/establishments/mainService.js [hunk 6]
```diff
@@ -39,53 +39,71 @@ router.route('/').get(async (req, res) => {
...
-
       // by loading after the restore, only those properties defined in the
       //  POST body will be updated (peristed)
       // With this endpoint we're only interested in name
-      const isValidEstablishment = await thisEstablishment.load({
-        mainService: req.body.mainService
-      });
-
+      let payload = {};
+      if (addIsRegulated){
```
Comment (chadtomkiss-madetech) [inline server/routes/establishments/mainService.js:70]: Using `if/else` to set a variable to the same thing but with added keys can be confusing.

You can add conditional keys to an object using spread syntax like this:

```
const payload =  {
  mainService: mainService,
  ...(addIsRegulated) && { isRegulated: true },
};
```

However, in this instance, I don't think you even need the `if/else`.

As long as `addIsRegulated` is always a boolean, you can rename `addIsRegulated` to `isRegulated` and do:

```
const payload = {
  mainService,
  isRegulated,
}
```

and it'll just update `isRegulated` to true/false as required.

Another option would be to have common keys at the initial assignment and then `if's` to add the conditional keys:

```
const payload = {
  mainService,
}

if (addIsRegulated) {
  payload.isRegulated = true;
}
```

File: server/test/unit/routes/admin/cqc-status-change/index.spec.js [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+const approvalStatus = (status) => {
+  returnedStatus = status;
+  return {json: approvalJson, send: () => {} };
+};
+var updateMainService;
+var throwErrorWhenFetchingAllRequests = false;
+var throwErrorWhenFetchingSingleRequest = false;
+
+
+describe.only('admin/cqc-status-change route', () => {
```
Comment (chadtomkiss-madetech) [inline server/test/unit/routes/admin/cqc-status-change/index.spec.js:77]: Mannn, I have this in my PRs so many times!

Remove `.only` :)

File: server/test/unit/routes/admin/cqc-status-change/index.spec.js [hunk 8]
```diff
@@ -0,0 +1,298 @@
...
+        } else if (id === fakeApproval.ID) {
+          return fakeApproval;
+        }
+      });
+
+      sb.stub(models.establishment, 'findbyId').callsFake(async (id) => {
+        if (id === testWorkplace.id) {
+          return testWorkplace;
+        }
+      });
```
Comment (chadtomkiss-madetech) [inline server/test/unit/routes/admin/cqc-status-change/index.spec.js:111]: Indentation issue here.

Hopefully one day we'll get Prettier running!

File: server/models/approvals.js [hunk 9]
```diff
@@ -66,7 +66,7 @@ module.exports = (sequelize, DataTypes) => {
         ApprovalType: approvalType,
         Status: 'Pending'
       },
-      attributes: ['ID', 'UUID', 'EstablishmentID', 'UserID', 'createdAt', 'Status'],
+      attributes: ['ID', 'UUID', 'EstablishmentID', 'UserID', 'createdAt', 'Status','Data'],
```
Comment (two-first-names) [inline server/models/approvals.js:69]: Can there be a space after the comma?

File: server/models/approvals.js [hunk 10]
```diff
@@ -90,7 +90,7 @@ module.exports = (sequelize, DataTypes) => {
       where: {
         ID: id
       },
-      attributes: ['ID', 'UUID', 'EstablishmentID', 'UserID', 'createdAt', 'Status'],
+      attributes: ['ID', 'UUID', 'EstablishmentID', 'UserID', 'createdAt', 'Status','Data'],
```
Comment (two-first-names) [inline server/models/approvals.js:93]: Ditto to the above.

File: server/routes/establishments/mainService.js [hunk 11]
```diff
@@ -39,53 +39,67 @@ router.route('/').get(async (req, res) => {
...
+
     if (err instanceof Establishment.EstablishmentExceptions.EstablishmentJsonException) {
       console.error("Establishment::mainService POST: ", err.message);
-      return res.status(400).send(err.safe);
+      return { success: false, errorMsg:err.safe,errorCode: 400 };
+
     } else if (err instanceof Establishment.EstablishmentExceptions.EstablishmentSaveException) {
       console.error("Establishment::mainService POST: ", err.message);
-      return res.status(503).send(err.safe);
+      return { success: false, errorMsg:err.safe,errorCode: 503 };
```
Comment (two-first-names) [inline server/routes/establishments/mainService.js:97]: Spaces here. (don't mind me, just want things to look nice!)

**Pull request merged**